fastlane_bot/events/managers/base.py

```diff
@@ -128,8 +129,10 @@ def __post_init__(self):
                 self.SUPPORTED_BASE_EXCHANGES.append(base_exchange_name)
 
             self.exchanges[exchange_name] = exchange_factory.get_exchange(key=exchange_name, cfg=self.cfg, exchange_initialized=initialize_events)
-            if base_exchange_name in SOLIDLY_V2_NAME:
+            if base_exchange_name == SOLIDLY_V2_NAME:
                 self.exchanges[exchange_name] = self.handle_solidly_exchanges(exchange=self.exchanges[exchange_name])
+            if base_exchange_name in [SOLIDLY_V2_NAME, UNISWAP_V2_NAME, UNISWAP_V3_NAME]:
+                self.exchanges[exchange_name] = self.initialize_factory_contract(exchange=self.exchanges[exchange_name])
```

Comment:
For `SOLIDLY_V2_NAME`, you are overriding the previous value of `self.exchanges[exchange_name]`.

So either the previous assignment is not needed, or this assignment is not needed.

Reply:
Fixed